test/plsql.test.ts

```diff
@@ -91,6 +91,15 @@ describe('PlSqlFormatter', () => {
     `);
   });
 
+  it.only('supports Q custom delimiter strings', () => {
+    expect(format("q'<test string < > 'foo' bar >'")).toBe("q'<test string < > 'foo' bar >'");
+    expect(format("Q'[test string [ ] 'foo' bar ]'")).toBe("Q'[test string [ ] 'foo' bar ]'");
+    expect(format("q'(test string ( ) 'foo' bar )'")).toBe("q'(test string ( ) 'foo' bar )'");
+    expect(format("Q'{test string { } 'foo' bar }'")).toBe("Q'{test string { } 'foo' bar }'");
+    expect(format("q'%test string % % 'foo' bar %'")).toBe("q'%test string % % 'foo' bar %'");
+    expect(format("Q'Xtest string X X 'foo' bar X'")).toBe("Q'Xtest string X X 'foo' bar X'");
```

Comment:
I suggest adding tests for code like:

```
q'X hello X'X''
```

To check that it gets tokenized as `q'X hello X'`, `X`, `''`.

Making sure that we don't greedily consume until last `X'`, but stop at the very first one.